src/app/groups/containers/user-group-invitations/user-group-invitations.component.ts

```diff
@@ -1,97 +1,122 @@
-        this.state = state.tag;
-        if (state.isReady) {
-          this.requests = state.data;
-        }
-      },
-      error: _err => {
-        this.state = 'fetchingError';
-      }
-    });
-  }
+    private processGroupInvitationService: ProcessGroupInvitationService,
+  ) {}
 
   ngOnInit(): void {
-    this.dataFetching.next({ sort: this.currentSort });
+    this.groupInvitationsSubject.next({ sort: this.currentSort });
   }
 
   ngOnDestroy(): void {
-    this.dataFetching.complete();
-  }
-
-  onProcessRequests(params: { data: PendingRequest[], type: Action }): void {
-    this.state = 'processing';
-    this.requestActionService.processGroupInvitations(params.data.map(r => r.group.id), params.type)
-      .subscribe({
-        next: result => {
-          this.state = 'ready';
-          displayResponseToast(this.actionFeedbackService, parseGroupInvitationResults(result), params.type);
-          this.dataFetching.next({ sort: this.currentSort });
-          this.currentContentService.forceNavMenuReload();
-        },
-        error: err => {
-          this.state = 'ready';
-          this.actionFeedbackService.unexpectedError();
-          if (!(err instanceof HttpErrorResponse)) throw err;
-        }
-      });
+    this.groupInvitationsSubject.complete();
   }
 
   onFetch(sort: string[]): void {
     if (JSON.stringify(sort) !== JSON.stringify(this.currentSort)) {
       this.currentSort = sort;
-      this.dataFetching.next({ sort: this.currentSort });
+      this.groupInvitationsSubject.next({ sort: this.currentSort });
     }
   }
+
+  onCustomSort(event: SortEvent): void {
+    const sortMeta = event.multiSortMeta?.map(meta => (meta.order === -1 ? `-${meta.field}` : meta.field));
+    if (sortMeta) this.onFetch(sortMeta);
+  }
+
+  onAccept(pendingRequest: PendingRequest): void {
+    this.processing = true;
+    this.processGroupInvitationService.accept(pendingRequest.group.id).subscribe({
+      next: result => {
+        this.processing = false;
+        if (!result.changed) {
+          this.actionFeedbackService.error(`The ${ pendingRequest.group.name } group has not been accepted`);
+          return;
+        }
+        this.actionFeedbackService.success(`The ${ pendingRequest.group.name } group has been accepted`);
+        this.groupInvitationsSubject.next({ sort: this.currentSort });
+        this.currentContentService.forceNavMenuReload();
+      },
+      error: err => {
+        this.processing = false;
+        this.actionFeedbackService.unexpectedError();
+        if (!(err instanceof HttpErrorResponse)) throw err;
+      },
+    });
+  }
+
+  onReject(pendingRequest: PendingRequest): void {
+    this.processing = true;
+    this.processGroupInvitationService.reject(pendingRequest.group.id).subscribe({
+      next: result => {
+        this.processing = false;
+        if (!result.changed) {
+          this.actionFeedbackService.success(`The ${ pendingRequest.group.name } group has been declined`);
```

Comment:
Typo: "has not" - I will fix it after review

Reply:
And probably this is error (not success message)? I didn't find test case for it

Reply:
`Unable to reject invitation to group "..."`
Yes, that should be a red error message

Reply:
Fixed